core/src/test/java/AllCoreUnitTestSuite.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco;
+
+import org.junit.experimental.categories.Categories;
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+
+@RunWith(Categories.class)
+@Suite.SuiteClasses({
+        org.alfresco.config.SystemPropertiesSetterBeanTest.class,
+        org.alfresco.encryption.AlfrescoKeyStoreTest.class,
+        org.alfresco.encryption.EncryptingOutputStreamTest.class,
+        org.alfresco.error.AlfrescoRuntimeExceptionTest.class,
+        org.alfresco.query.CannedQueryTest.class,
+        org.alfresco.util.collections.CollectionUtilsTest.class,
+        org.alfresco.util.exec.ExecParameterTokenizerTest.class,
+        org.alfresco.util.exec.RuntimeExecBeansTest.class,
+        org.alfresco.util.exec.RuntimeExecTest.class,
+        org.alfresco.util.random.NormalDistributionHelperTest.class,
+        org.alfresco.util.shard.ExplicitShardingPolicyTest.class,
+        org.alfresco.util.transaction.SpringAwareUserTransactionTest.class,
+        org.alfresco.util.BridgeTableTest.class,
+        org.alfresco.util.CachingDateFormatTest.class,
+        org.alfresco.util.DynamicallySizedThreadPoolExecutorTest.class,
+        org.alfresco.util.EqualsHelperTest.class,
+        org.alfresco.util.GuidTest.class,
+        org.alfresco.util.ISO8601DateFormatTest.class,
+        org.alfresco.util.LogAdapterTest.class,
+        org.alfresco.util.LogTeeTest.class,
+        org.alfresco.util.PathMapperTest.class,
+        org.alfresco.util.TempFileProviderTest.class,
+        org.alfresco.util.VersionNumberTest.class
+})
+public class AllCoreUnitTestSuite {
```

Comment:
I noticed there are several new TestSuites added as part of this PR.

For each TestSuite can you explain:
- how they were generated (e.g.: reverse-engineered from GHA logs/Surefire reports)
- when we'll need to update them (e.g.: update AllCoreUnitTestSuite if you need a new unit test to be executed as part of https://github.com/Alfresco/alfresco-community-repo/pull/2445/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR159-R161)
- since these tests already existed and they were just grouped by a TestSuite, how come we didn't have to modify any existing TestSuite e.g.: removing the tests that have now been moved to this new TestSuite? Were all the tests for each of these test suites never part of any other test suite before?

Thanks 🙏

Reply:
I still need answers to these questions @wojtekswieton

Reply:
>-how they were generated (e.g.: reverse-engineered from GHA logs/Surefire reports)

I created 3 new test suites.
AllCoreUnitTestSuite and AllDataModelUnitTestSuite were necessary after split 'Core, Data-Model, Repository - AllUnitTestsSuite - Build and test' to 4 jobs.
Before that tests in that job were run by mixing 2 different ways:
- name: "Run tests" run: | mvn -B test -pl core,data-model -am -DfailIfNoTests=false mvn -B test -pl "repository,mmt" -am "-Dtest=AllUnitTestsSuite,AllMmtUnitTestSuite" -DfailIfNoTests=false
Repository tests and mmt - by directly calling test suite. Core, data-model by using surefire test including mechanism in core and data-model modules. So I created a test suites for them - and add all tests located inside those modules and also manually checked on GHA logs if all of them are added to test suites. Right now all tests in those 4 modules are runs by the same way.

Third new test suite: MessagingUnitTestSuite just grouped together 2 test classes that were before runs separately which generates 2 reports.
run: mvn -B test -pl repository -am -Dtest=CamelRoutesTest,CamelComponentsTest -DfailIfNoTests=false 

Most details about those new test suites I describe in a confluence page: https://hyland.atlassian.net/wiki/spaces/TECH/pages/1336050823/alfresco-community-repo+integration+with+Report+Portal

>-when we'll need to update them (e.g.: update AllCoreUnitTestSuite if you need a new unit test to be executed as part of https://github.com/Alfresco/alfresco-community-repo/pull/2445/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR159-R161)

When we need to add a new test cases or update just need to check if there are added to correct test suite.

>-since these tests already existed and they were just grouped by a TestSuite, how come we didn't have to modify any existing TestSuite e.g.: removing the tests that have now been moved to this new TestSuite? Were all the tests for each of these test suites never part of any other test suite before?

As I wrote above - those tests in new test suites weren't in any test suite before. They were runs by surefire mechanism or were directly indicated.